### DISCUSSION TITLE
SREFLEX error: ERROR 116 FILE 'initial001\initial.vectors' NOT FOUND.

### DISCUSSION THREAD

**Discussion by ZhihB** [biosaxs-com/atsas-community#1931] [Mixtures and flexible systems] [answered]
Hi all,

I am a beginner with ATSAS and plan to use SREFLEX to generate models for scattering data. 
I was running SREFLEX with 5 defined-domains using the PowerShell script:

C:\ATSAS-3.2.1\bin> .\sreflex.exe --prefix=Run --skip=UNRESTRAINED --threads=6 chromixs01.dat Domain1.pdb,Domain2.pdb,Domain3.pdb,Domain4.pdb,Domain5.pdb

And I got this error message:

--NMA: Running initial normal mode analysis...
ERROR 116 FILE 'initial001\initial.vectors' NOT FOUND.

When I deleted Domain5.pdb from the script (see below), SREFLEX ran successfully.

C:\ATSAS-3.2.1\bin> .\sreflex.exe --prefix=Run --skip=UNRESTRAINED --threads=6 chromixs01.dat Domain1.pdb,Domain2.pdb,Domain3.pdb,Domain4.pdb

I still wish to include the Domain5.pdb for model generation. I wonder if there is a solution for this issue. 
Also, what does ERROR 116 actually mean? Was there a problem with the Domain5.pdb? Or was there a problem with my setting? Thank you so much!

Best wishes
Zhi

**Comment by maxim-biosaxs** [comment]
Possibly, you mistyped the name of the last file.
Please try 
Domain5.pdb 
instead of
Domain5,pdb

**Reply by ZhihB** [reply]
Hi Maxim,

Thanks a lot for your suggestion. I used Domain5.pdb in my original script (Domain5,pdb was a typo in the discussion and I have edited it, thanks) and it still showed the same error 116. Do you have other suggestions? 
Thank you so much and sorry for taking your time.

Best wishes,
Zhi

**Comment by franke-biosaxs** [comment]
Hi Zhi.

Did you try with the latest [ATSAS-4.0.0](https://github.com/biosaxs-com/atsas-community/releases)? There weren't many changes in sreflex since 3.2.1, but if there are problems with Domain5.pdb specifically, a better error message might be provided. If an update does not help, would you be able to share the files so I can have a more detailed look?

Regards

Daniel

**Reply by ZhihB** [reply]
Hi Daniel,

Thanks for suggesting to try ATSAS-4.0.0. I got the same error message "ERROR 116 FILE 'initial001\initial.vectors' NOT FOUND.". Unfortunately, no other details were provided. 

The scatter data and the pdb files are attached. I would be very grateful if you could have a more detailed look. Thank you so much!
[Sreflex.zip](https://github.com/user-attachments/files/16589197/Sreflex.zip)


P.S. I tried running SREFLEX for the scatter data with Domain5.pdb alone, and it ran smoothly. 

Best wishes,
Zhi

**Comment by franke-biosaxs** [comment, marked as answer]
So. I had a look. And I have to admit that I am a bit at a loss.

First I confirmed the issue:
```
$ sreflex --prefix=run --skip=UNRESTRAINED --threads=8 chromixs01.dat Domain1.pdb,Domain2.pdb,Domain3.pdb,Domain4.pdb,Domain5.pdb
[...]
--NMA: Running initial normal mode analysis...
ERROR 116 FILE 'initial001/initial.vectors' NOT FOUND.
```
This error is thrown after the run of a utility external to `sreflex`, namely `diagrtb`. One can run this individually in the `runXXX/initial001` directory after `sreflex` stops. Then one obtains:

```
$ cd run003/initial001/
$ diagrtb                  
 Diagrtb> Diagonalizes a matrix, using the RTB approximation.
 Diagrtb> Version 2.64, November 2011. 
 Diagrtb> Options to be read in diagrtb.dat file.

 Diagrtb> Main options:

 MATRix filename          = initialwrap.sdijf
 COORdinates filename     = initialwrap.coor
 Eigenvector OUTPut file  = initial.vectors
 Nb of VECTors required   =         64
 EigeNVALues chosen       =       LOWE
 Nb of residues per BLOck =          3
 Origin of MASS values    =       CONS

 Blocpdb>   386 atoms picked in pdb file.
 Blocpdb>  All masses set to unity.
 Blocpdb> Last block contains only  1 residue(s).

%Blocpdb-Wn> 1 atom(s) in block   64 i.e., less than what is required for a rigid body.
%Blocpdb-Wn> Last atom found for this block is the    190th, in residue A   944
%Blocpdb-Wn> It will be included in the next one.

%Blocpdb-Wn> 2 atom(s) in block  102 i.e., less than what is required for a rigid body.
%Blocpdb-Wn> Last atom found for this block is the    306th, in residue A   527
%Blocpdb-Wn> It will be included in the next one.

 Blocpdb>   127 blocks.

 Prepmat> Matrix trace    =       6781.0000

%Prepmat-Er>  1110 coordinates found in this matrix, 
 instead of    386*3, as expected. Not the right one ?
STOP *Wrong matrix or wrong coordinate file*
```

To note, the input of `diagrtb` is obtained by running `pdbmat` beforehand. 

Looking at the initial input file for `pdbmat`, i.e. `initialwrap.coor`, I notice that the order of residues and the atom numbering are not in sequence. This is an educated guess, but assuming that the domains should be consecutive along the chain, I would start by straightening out the atom and residue numbering such that the five input domains result in a correctly numbered sequence and try again.  Looking closer, the two messages about "A 944" and "A 527" are where the residue numbers drop, i.e. 944 followed by 412 and 527 followed by 178. While at it, maybe also provide appropriate chain IDs?

If the above doesn't help, a more detailed look into your domains would be required. Something in there doesn't match the expectation of the utility tools.

HTH.

**Reply by ZhihB** [reply]
Hi Daniel,

Thank you so much for looking at the files and test-running Sreflex with them! I followed your suggestion and changed the residue numbers of the domains so they now form the correctly numbered sequence. I am currently running Sreflex with these new pdb files and Sreflex started processing successfully. Thanks again!

Best wishes,
Zhi
